packages/block-editor/src/hooks/duotone.js

```diff
@@ -121,9 +123,9 @@ function DuotonePanel( { attributes, setAttributes } ) {
 		return null;
 	}
 
-	const duotonePresetOrColors = ! Array.isArray( duotoneStyle )
-		? getColorsFromDuotonePreset( duotoneStyle, duotonePalette )
-		: duotoneStyle;
+	const duotonePresetOrColors = duotonePreset
+		? getColorsFromDuotonePreset( duotonePreset, duotonePalette )
+		: duotoneColors;
```

Comment:
## TODO

Bug: the "unset" value is stored in the style attribute so we need a custom condition here to handle when duotone is hard value "unset" otherwise this code will retriebe the colors from the default duotone palette.